### DISCUSSION TITLE
Cronos EVM Mainnet Pallene v1.4 Upgrade Proposal

### DISCUSSION THREAD

**Discussion by lezzokafka** [crypto-org-chain/cronos#1707] [Ideas]
![image (34)](https://github.com/user-attachments/assets/698d5d0a-ab7d-41f4-8d18-5408924929a8)

# Introduction 
**Cronos EVM Pallene introduces the fastest transaction processing capabilities ever seen on the network, by a wide margin.** With parallel transaction processing and optimistic execution, the virtual machine acquires the superpower of being able to deliver tens of thousands of transactions per second (compared with up to 100 TPS today).

Parallel transaction processing has been a major narrative among Layer 1 networks recently, and this evolution makes Cronos EVM future proof as the Web3 industry enters a new cycle of mainstream adoption. 

Before hyper-fast TPS can be seen live on the [blockchain explorer](https://explorer.cronos.org/), more changes need to happen: first, the time between each block must be agreed to be set to a shorter interval (currently, 5 to 6 seconds); second, the volume of user transactions needs to become large enough that the cost of faster blocks becomes economically sustainable for network validators. Nevertheless, Pallene is a revolutionary step on Cronos' scalability journey.

Cronos EVM Pallene also includes multiple upgrades leveraging the latest Cosmos SDK and CometBFT (formerly "Tendermint") innovations while optimizing them at the same time.

Please read below to learn more about today's governance discussion.

# Block-STM: Parallel Transaction Processing 

## Overview of Block-STM

The introduction of block-stm ("block-level Software Transactional Memory") redefines how transactions are processed on Cronos.

Moving from sequential to concurrent processing, this approach enables multiple transactions to be processed within each block at once. Leveraging software transactional memory (STM), this framework efficiently manages concurrent transactions by automatically resolving conflicts and reducing latency.

With block-stm, Cronos achieves significantly higher transaction throughput, meeting the demands of performance-driven apps and services.

## Benchmarking Results

The latest release brings a new benchmark framework that developers can configure in JSON format in order to compare the virtual machine's performance across various software versions and simulation environments such as laptops, Docker containers and Kubernetes clusters.

Naturally, this benchmarking framework was used to [measure the performance improvements brought by Block-STM](https://github.com/crypto-org-chain/cronos/wiki/V1.4-Benchmark). The chart below shows how the introduction of Block-STM enables Cronos EVM to execute up to 60,000 transactions per second (vs up to 100 TPS today).
![VALIDATORS v2](https://github.com/user-attachments/assets/0219635d-a1ec-42c0-8bd0-f9b93e227f97)

The lower performance of single transfers, compared with batched transfers, actually reveals a limitation of the load generator of the benchmarking framework, which struggled to feed transactions as fast as the network could execute them! This will be addressed in future versions of the benchmarking framework.

## Significant Reduction of Network Overhead

This upgrade brings practical improvements to reduce network overhead by refining the EVM message format and optimizing the Ethermint ante handler.

The updated message format simplifies how transactions are encoded and decoded, making them faster to process. The redesigned ante handler cuts out redundant steps, improves caching for signature checks, and optimizes fee and gas validation.

These changes work together to handle transactions more efficiently, improving the speed and performance of the entire network. 

## Optimization of Cosmos SDK and CometBFT

Custom enhancements were made to the Cosmos SDK and CometBFT layers to improve efficiency and performance.

A new mechanism was introduced to store Golang objects directly rather than serialized bytes, which reduces the computational cost of encoding and decoding

The fee collection logic was refined to support parallel transaction execution which enables faster and more efficient handling of transactions under high load.

Event firing can now happen asynchronously in CometBFT, moving a significant portion of the workload out of the block commit process.

These improvements speed up block processing and enhance the network’s ability to handle transaction surges. 

## Cosmos SDK and IBC Upgrades

The Pallene upgrade introduces important new features by upgrading Cosmos SDK and IBC dependencies.

### Optimistic Execution (SDK 0.50)
The integration of Optimistic Execution from Cosmos SDK 0.50 represents a major step forward in transaction processing.

Traditionally, a block proposal is processed only after voting concludes. With ABCI++, the application layer can start processing the proposal in advance. By executing block proposals in parallel, Optimistic Execution can significantly cut down block times, enhancing the network’s responsiveness and capacity, particularly in high-demand scenarios.

### Improved Interchain Accounts Stability with IBC v8.5.2

IBC v8.5.2 introduces support for [unordered interchain-accounts channels](https://ibc.cosmos.network/v7/apps/interchain-accounts/active-channels/), making these channels more resilient and practical for live networks.

This improvement strengthens Cronos's cross-chain compatibility by allowing for more stable and flexible interchain accounts operations.

## Bug Fixes and Others Dependency Upgrades

Various bug fixes and enhancements have been introduced to further stabilize the network and improve the developer experience.

These adjustments collectively contribute to the overall reliability of the Cronos EVM chain. For a detailed breakdown of these changes, refer to the [Cronos GitHub changelog](https://github.com/crypto-org-chain/cronos/blob/v1.4.0-rc3/CHANGELOG.md).

# Looking Ahead

The Cronos EVM v1.4.0 Pallene upgrade lays a strong foundation for the platform’s future.

While Cronos zKEVM, newly launched in 2024, is the best blockchain network for users and developers seeking full integration with Ethereum and its Layer 2s together with the decentralized security of Ethereum, the Cronos EVM Layer 1 retains significant room to grow. It is extremely well placed to compete for use cases that require hyper-fast transaction processing (e.g., TradFi, payments, Web3 games, NFTs).

The proposed upgrade marks another step toward Cronos’ vision of building a complete, reliable, high-performance blockchain ecosystem.

For more details and updates, visit [Cronos.org](http://cronos.org/) and subscribe to the [Cronos newsletter](https://blog.cronos.org/).

**Comment by Th0rgal** [comment]
Hey! Cool upgrade on Pallene. Quick question: Any plans to make gas fees more dynamic with the new TPS capacity?

**Comment by Undead8** [comment]
Will the block height at which the upgrade will happen be posted here or in another discussion?
